File: server/src/main/java/bit/project/server/entity/Customer.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    private String name;

    private String mobile;

    private String land;

    private String address;

    private String email;

    private String fax;

    @ManyToOne
    @JsonIgnoreProperties({"creator", "status", "tocreation", "rolelist"})
    private User creator;

    public Customer(Integer id){
        this.id=id;
    }

    public Customer(Integer id, String code,String name) {
        this.id = id;
        this.code = code;
        this.name=name;
    }
}

File: client/src/app/entities/customer.ts
import {User} from './user';
import {DataPage} from '../shared/data-page';

export class Customer{
  id: number;
  code: string;
  tocreation: string;
  description: string;
  name: string;
  mobile: string;
  land: string;
  address: string;
  email: string;
  fax: string;
  creator: User;


  constructor(id: number= null) {
    this.id = id;
  }
}



export class CustomerDataPage extends DataPage{
  content: Customer[];
}
